### PR TITLE
[DOCS] Adding authorization_realms setting for OIDC

### LINKED ISSUE #64583: OpenId realm supports the authorization_realms setting
**Description of the problem including expected versus actual behavior**:
This is a documentation issue about OpenID realm not showing  `authorization_realms` setting support.

**Steps to reproduce**:

In the  [security settings](https://www.elastic.co/guide/en/elasticsearch/reference/7.9/security-settings.html#ref-oidc-settings), the `authorization_realms` setting is missing from the Open ID connect realm.
I do see the setting 4 times for the following realms: SAML, LDAP, PKI and Kerberos.
This implies this `authorization_realms` setting is not valid for OpenID realm hence authorization delegation is not supported.

However the [role mapping page](https://www.elastic.co/guide/en/elasticsearch/reference/current/oidc-role-mapping.html#oidc-role-mapping) shows this setting is in fact supported.

- Could we add edit the [security settings](https://www.elastic.co/guide/en/elasticsearch/reference/7.9/security-settings.html#ref-oidc-settings) to add documenation about `authorization_realms` setting for OpenID?

**Provide logs (if relevant)**:

### PR DESCRIPTION
**Pull request by lockewritesdocs**
Adds the `authorization_realms` setting for OpenID Connect, and updates mentions that OpenID Connect supports this setting.

image

Closes #64583

### CODE REVIEW

File: x-pack/docs/en/security/authentication/oidc-guide.asciidoc [hunk 1]
```diff
@@ -483,12 +483,13 @@ If your users also exist in a repository that can be directly accessed by {es}
 <<authorization_realms, authorization realms>> instead of role mappings.
 
 In this case, you perform the following steps:
+
 1. In your OpenID Connect realm, assign a claim to act as the lookup userid,
    by configuring the `claims.principal` setting.
-2. Create a new realm that can lookup users from your local repository (e.g. an
+2. Create a new realm that can lookcup users from your local repository (e.g. an
```
Comment (lcawl) [inline x-pack/docs/en/security/authentication/oidc-guide.asciidoc:489]:
```suggestion
2. Create a new realm that can look up users from your local repository (e.g. an
```

File: x-pack/docs/en/security/authorization/mapping-roles.asciidoc [hunk 2]
```diff
@@ -10,6 +10,9 @@ For other types of realms, you must create _role-mappings_ that define which
 roles should be assigned to each user based on their username, groups, or
 other metadata.
 
+NOTE: When <<anonymous-access,anonymous access>> is enabled, the roles
```
Comment (jkakavas) [inline x-pack/docs/en/security/authorization/mapping-roles.asciidoc:13]: These are two unrelated statements that both fit a note section. I think that merging them make doesn't make much sense.

**Review by jkakavas — CHANGES_REQUESTED** [review]

File: x-pack/docs/en/security/authorization/mapping-roles.asciidoc (hunk 2 above)
Comment (lockewritesdocs) [reply x-pack/docs/en/security/authorization/mapping-roles.asciidoc:13]: Thanks @jkakavas -- I think you're right. I split the notes out into separate notes and moved the one about users with no roles after the next paragraph.

**Review by jkakavas — APPROVED** [review]
LGTM, thanks @lockewritesdocs !

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Pull request merged**